base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs

```diff
@@ -441,11 +539,75 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
             UpdateOutput {
                 status: Some(OutputStatus::Unspent),
                 tx_id: None,
+                spending_key: None,
             },
             &(*conn),
         )?;
         Ok(())
     }
+
+    fn apply_encryption(&self, cipher: Aes256Gcm) -> Result<(), OutputManagerStorageError> {
+        let mut current_cipher = acquire_write_lock!(self.cipher);
+
+        if (*current_cipher).is_some() {
+            return Err(OutputManagerStorageError::AlreadyEncrypted);
+        }
+
+        let conn = acquire_lock!(self.database_connection);
+        let mut outputs = OutputSql::index(&conn)?;
+
+        // If the db is already encrypted then the very first output we try to encrypt will fail.
+        for o in outputs.iter_mut() {
+            // Test if this output is encrypted or not to avoid a double encryption.
+            let _ = PrivateKey::from_vec(&o.spending_key).map_err(|_| {
+                error!(
+                    target: LOG_TARGET,
+                    "Could not create PrivateKey from stored bytes, They might already be encrypted"
+                );
+                OutputManagerStorageError::AlreadyEncrypted
+            })?;
+            o.encrypt(&cipher)
+                .map_err(|_| OutputManagerStorageError::AeadError("Encryption Error".to_string()))?;
+            o.update_encryption(&conn)?;
+        }
+
+        let mut key_manager_state = KeyManagerStateSql::get_state(&conn)?;
+        key_manager_state
```

Comment:
I might be mistaken, but it seems like if there are no outputs, the key_manager_state could get encrypted twice.

Reply:
This is true, lemme add the check quick.